### PR TITLE
Removed JUnit5 dependency for ClientTpcConfigTest

### PR DESCRIPTION
**Pull request by dr-divago**
Removed JUnit5 dependency for ClientTpcConfigTest

Fixes [#26193](https://github.com/hazelcast/hazelcast/issues/26193)

Checklist:
- [ ] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Add `Add to Release Notes` label if changes should be mentioned in release notes or `Not Release Notes content` if changes are not relevant for release notes
- [x] Request reviewers if possible
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases

### CODE REVIEW

**Comment by JackPGreen** [comment]
@dr-divago thanks for your contribution!

It's great you've addressed these issues - had you considered amending the test referenced in https://github.com/hazelcast/hazelcast/issues/26193 to see if further examples could be caught going forwards?

**Comment by dr-divago** [comment]
@JackPGreen Do you mean to check all the references of JUnit5 in the codebase? Or fixing ```com.hazelcast.NoMixedJUnitAnnotationsInOurTestSourcesTest.noJUnitMixing()?```

**Comment by JackPGreen** [comment]
> @JackPGreen Do you mean to check all the references of JUnit5 in the codebase? Or fixing `com.hazelcast.NoMixedJUnitAnnotationsInOurTestSourcesTest.noJUnitMixing()?`

I thought fixing the `noJUnitMixing` test to find other scenarios and ensure this doesn’t happen again.

**Comment by dr-divago** [comment]
Hi @JackPGreen 
I have added chehck for assert in noJUnitMixing and including IT tests in the scanning

**Comment by JackPGreen** [comment]
The job `Hazelcast-mono-OS-pr-builder` of your PR failed. (Hazelcast internal details: [build log](https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-mono-OS-pr-builder/3132/Hazelcast-mono-OS-pr-builder-3132.zip), [artifacts](https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-mono-OS-pr-builder/3132/Hazelcast-mono-OS-pr-builder-3132.zip)).
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.3.1:checkstyle (default) on project hazelcast-archunit-rules: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There are 2 errors reported by Checkstyle 9.3 with /home/jenkins/jenkins_slave/workspace/Hazelcast-mono-OS-pr-builder_2/hazelcast/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-mono-OS-pr-builder_2/hazelcast/hazelcast-archunit-rules/src/main/java/com/hazelcast/test/archunit/MixTestAnnotationsCondition.java:24:17: Using the '.*' form of import should be avoided - org.junit.*. [AvoidStarImport]
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-mono-OS-pr-builder_2/hazelcast/hazelcast-archunit-rules/src/main/java/com/hazelcast/test/archunit/MixTestAnnotationsCondition.java:39:59: Name 'JUNIT_4_ClASS_ANNOTATION' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [ConstantName]
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.3.1:checkstyle (default) on project hazelcast-archunit-rules: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There are 2 errors reported by Checkstyle 9.3 with /home/jenkins/jenkins_slave/workspace/Hazelcast-mono-OS-pr-builder_2/hazelcast/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
[ERROR] 
--------------------------
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
--------------------------
[ERROR] 
--------------------------
[ERROR] For more information about the errors and possible solutions, please read the following articles:
--------------------------
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
--------------------------
[ERROR] 
--------------------------
[ERROR] After correcting the problems, you can resume the build with the command
--------------------------
[ERROR]   mvn <args> -rf :hazelcast-archunit-rules
--------------------------
</pre>
</details>

**Comment by dr-divago** [comment]
Hi @JackPGreen
I updated the PR. How to access the build log?

**Comment by JackPGreen** [comment]
run-os-tests
run-ee-tests

**Comment by JackPGreen** [comment]
> Hi @JackPGreen I updated the PR. How to access the build log?

It's an internal tool unfortunately, but all tests have passed :)

File: hazelcast-archunit-rules/src/main/java/com/hazelcast/test/archunit/MixTestAnnotationsCondition.java [hunk 1]
```diff
@@ -51,18 +63,45 @@ public MixTestAnnotationsCondition() {
...
+    }
+
+    private boolean hasJUnit5Assertions(JavaClass item) {
+        Set<JavaMethodCall> methodCalls = item.getMethodCallsFromSelf();
+        return methodCalls.stream().anyMatch(call -> call.getTarget().getFullName().contains("org.junit.jupiter.api.Assertions"));
+    }
+    private boolean hasJUnit4Assertions(JavaClass item) {
+        Set<JavaMethodCall> methodCalls = item.getMethodCallsFromSelf();
+        return methodCalls.stream().anyMatch(call -> call.getTarget().getFullName().contains("org.junit.Assert"));
+    }
```
Comment (JackPGreen) [inline hazelcast-archunit-rules/src/main/java/com/hazelcast/test/archunit/MixTestAnnotationsCondition.java:82]: nit - it's possible to centralise the logic into a single method that these can delegate to with different `fullName` parameters

File: hazelcast-archunit-rules/src/main/java/com/hazelcast/test/archunit/MixTestAnnotationsCondition.java [hunk 2]
```diff
@@ -51,18 +63,45 @@ public MixTestAnnotationsCondition() {
...
+        boolean hasJUnit4Annotation = item.getMethods().stream()
+                .anyMatch(method -> JUNIT_4_ANNOTATION_CLASSES.stream()
+                        .anyMatch(method::isAnnotatedWith));
+
+        boolean hasJUnit4ClassAnnotation = item.getAnnotations().stream()
+                .anyMatch(annotation -> JUNIT_4_CLASS_ANNOTATION.stream()
+                        .anyMatch(aa -> annotation.getRawType().isAssignableFrom(aa)));
+
+        return hasJUnit4Annotation || hasJUnit4ClassAnnotation;
     }
```
Comment (JackPGreen) [inline hazelcast-archunit-rules/src/main/java/com/hazelcast/test/archunit/MixTestAnnotationsCondition.java:105]: Same here.

File: hazelcast-archunit-rules/src/main/java/com/hazelcast/test/archunit/MixTestAnnotationsCondition.java [hunk 3]
```diff
@@ -51,18 +63,45 @@ public MixTestAnnotationsCondition() {
...
                 .anyMatch(method -> JUNIT_5_ANNOTATION_CLASSES.stream()
                         .anyMatch(method::isAnnotatedWith));
 
-        if (hasJUnit4Annotation && hasJUnit5Annotation) {
-            String message = String.format("Class %s mixes JUnit 4 and JUnit 5 annotations.", item.getName());
-            events.add(SimpleConditionEvent.violated(item, message));
-        }
+        return hasJUnit5Annotation || hasJUnit5ClassAnnotation;
+    }
+    private boolean hasAnyJUnit4Annotations(JavaClass item) {
```
Comment (JackPGreen) [inline hazelcast-archunit-rules/src/main/java/com/hazelcast/test/archunit/MixTestAnnotationsCondition.java:95]:
```suggestion
    }

    private boolean hasAnyJUnit4Annotations(JavaClass item) {
```

Add some space between methods

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/SqlPrunedMemberLeftTest.java
Comment (JackPGreen) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/SqlPrunedMemberLeftTest.java:1]: Thanks for finding and fixing these!

File: hazelcast/src/test/java/com/hazelcast/client/config/ClientTpcConfigTest.java [hunk 4]
```diff
@@ -24,10 +24,12 @@
...
 import org.junit.runner.RunWith;
 
-import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.assertThrows;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
```
Comment (JackPGreen) [inline hazelcast/src/test/java/com/hazelcast/client/config/ClientTpcConfigTest.java:33]:
```suggestion

```

I don't think we need these extra blank lines?

**Review by JackPGreen — CHANGES_REQUESTED** [review]

**Comment by dr-divago** [comment]
Hi @JackPGreen is something missing in the PR?

**Comment by JackPGreen** [comment]
> @JackPGreen why the PR was closed?

It’s open again now, apologies :)

File: hazelcast/src/test/java/com/hazelcast/internal/util/StringUtilTest.java [hunk 5]
```diff
@@ -20,8 +20,8 @@
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
```
Comment (TomaszGaweda) [inline hazelcast/src/test/java/com/hazelcast/internal/util/StringUtilTest.java:23]: In this form it won't work:
- Junit 4 test must be public
- We use also ParameterizedTest in this class

**Review by TomaszGaweda — CHANGES_REQUESTED** [review]

File: hazelcast/src/test/java/com/hazelcast/internal/util/StringUtilTest.java (hunk 5 above)
Comment (dr-divago) [reply hazelcast/src/test/java/com/hazelcast/internal/util/StringUtilTest.java:23]: You are right, I miss it. Do you suggest rewrite the test with Parameterized runner?

**Pull request closed without merging**